/*
Basic MODBUS communication tool
Copyright (C) 2017 Martin ZÃºber

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software Foundation,
Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
*/

#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>
#include <QSerialPort>
#include <QSerialPortInfo>

namespace Ui {
class MainWindow;
}

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    explicit MainWindow(QWidget *parent = 0);
    ~MainWindow();

private slots:
    void on_serialPortReadyRead();
    void on_serialPortError(QSerialPort::SerialPortError err);
    void on_action_Quit_triggered();
    void on_action_about_triggered();

private:
    Ui::MainWindow *ui;

    QSerialPort port;
    QByteArray readBuffer;

    void initializePort();
    bool validator();
    void info(const QString& msg);
};

#endif // MAINWINDOW_H
